modules/mono/csharp_script.cpp

```diff
@@ -2445,6 +2446,8 @@ RES ResourceFormatLoaderCSharpScript::load(const String &p_path, const String &p
 	if (r_error)
 		*r_error = OK;
 
+	Ref<CSharpScript> scriptres(script);
+
 	return scriptres;
 }
```

Comment:
`script` would be leaked if the `ERR_FAIL_COND_V` is triggered, so we need to `memdelete` it.
But #21682 shows that it crashes if you delete `script` while `scriptres` references it, so since nothing needs `scriptres` until the `return scriptres;`, I moved its definition down. An alternative would be `scriptres->unref();` in the `if (err != OK)` I guess.

Reply:
But why is it leaked in the first place? If `ERR_FAIL_COND_V` is triggered, `script` should be deleted by the `scriptres` destructor, since it's the only one referencing it.

Reply:
The problem is with the GDScript code memdeleting `script` in the first place, when it was already set to be freed by `scriptres` destructor.

Reply:
If I'm right, the fix should only require removing the two `memdelete(script);` from the GDScript version.

Reply:
Ah that's right, I'll do this then, thanks!